internal/server/iito.go

```diff
@@ -48,54 +48,61 @@ func (s *SOMASServer) runGiftSession() error {
 	return nil
 }
 
-// GetGiftRequests collects all gift requests from the clients in a map
-func (s *SOMASServer) getGiftRequests() map[shared.ClientID]shared.GiftRequestDict {
-	giftRequestDict := map[shared.ClientID]shared.GiftRequestDict{}
+// GetGiftRequests collects a sorted list of gift requests from an individual client, for all clients, in a map
+func (s *SOMASServer) getGiftRequests() map[shared.ClientID][]shared.GiftRequest {
+	giftRequestDict := map[shared.ClientID][]shared.GiftRequest{}
 	for id, client := range s.clientMap {
-		giftRequestDict[id] = client.GetGiftRequests()
+		giftRequestDict[id] = shared.SortGiftRequestByTeam(client.GetGiftRequests())
 	}
 	return giftRequestDict
 }
 
-// GetGiftOffers collects all responses from clients to their requests in a map
-func (s *SOMASServer) getGiftOffers(totalRequests map[shared.ClientID]shared.GiftRequestDict) (map[shared.ClientID]shared.GiftOfferDict, error) {
-	totalOffers := map[shared.ClientID]shared.GiftOfferDict{}
-	// Loop over each team
+// getGiftOffers collects all responses from clients to their requests in a map
+func (s *SOMASServer) getGiftOffers(totalRequests map[shared.ClientID][]shared.GiftRequest) (map[shared.ClientID][]shared.GiftOffer, error) {
+	totalOffers := map[shared.ClientID][]shared.GiftOffer{}
 	for id, client := range s.clientMap {
 		// Gather all the requests made to this team
-		requestsToThisTeam := shared.GiftRequestDict{}
+		requestsToThisTeam := []shared.GiftRequest{}
 		for fromTeam, indivRequests := range totalRequests {
+			// TODO: Will fail if the list does not have all 6 entries. Fix in a sanitisation function.
 			requestsToThisTeam[fromTeam] = indivRequests[id]
 		}
+
 		offer, err := client.GetGiftOffers(requestsToThisTeam)
-		if err != nil {
-			totalOffers[id] = offer
+		if err == nil {
```

Comment:
Wouldn't this mean there is no error, causing the function to return and asking no other clients for their offers?